Algorithm.Framework/Selection/UncorrelatedUniverseSelectionModel.py

```diff
@@ -76,12 +76,13 @@ def SelectCoarse(self, algorithm, coarse):
             if not data.IsReady:
                 newSymbols.append(symbol)
 
-        # Warm up the dictionary objects of selected symbols and benchmark that do not have enought data
+        # Warm up the dictionary objects of selected symbols and benchmark that do not have enough data
         if len(newSymbols) > 1:
             history = algorithm.History(newSymbols, self.historyLength, Resolution.Daily)
-            history = history.close.unstack(level=0)
-            for symbol in newSymbols:
-                self.cache[symbol].Warmup(history)
+            if not history.empty:
+                history = history.close.unstack(level=0).fillna(method='bfill')
```

Comment:
We cannot use backward fill, `bfill`, to warm up indicators, because it will introduce look-ahead bias. Instead, we should use forward fill, `ffill`, and drop all remaining `nan`:
```python
history = history.close.unstack(level=0).ffill().dropna()
```

Reply:
Using dropna() reduces the number of rows in the DataFrame when data history is not available for all symbols, and it will not solve the problem because we need a complete history to calculate the Z-score.

It is unlikely that securities with values filled using 'bfill' will score high on Z-score as there is no fluctuation in correlation until data becomes available.

Reply:
We really should not use `bfill`.
If we don't have a complete history, maybe we should discard the symbol altogether (especially if it is unlikely to have high Z-score for it). 
Alternatively, we drop the `nan` from the symbol and only discard the symbol if x% was dropped. In this case, we will drop the same dates from the benchmark history as well to have arrays with the same size.

Reply:
There is no need to use `ffill` or `bfill`  as SymbolData returns 0 if it is not ready. Only condition needed to resolve the problem is `if not history.empty`.